### PR TITLE
Fix test discovery with duplicate transcript and key,value search

### PR DESCRIPTION
**Pull request by sv-github**
Issues fixed when using the integration tests with json input

ensure no duplicate 'transcript' key is present
do not search key,values in word_confusions

### CODE REVIEW

File: testing/evaluate_tests.py [hunk 1]
```diff
@@ -113,6 +113,24 @@ def test_schema(full_response, test_value, test_name=''):
...
+    # interpreted_fields:
+    #       company_name: '@nlprocessor_company_name'
+    #       key_metrics:
+    #         revenue:
+    #
+    # the revenue key metric was never found since the word confusion {'revenue':1} was found first
+
+    # assume there is only one segment since that's how the tests are written
+    if full_response.get('segments') and full_response.get('segments')[0].get('word_confusions'):
+        del full_response['segments'][0]['word_confusions']
```
Comment (cbrochtrup) [inline testing/evaluate_tests.py:132]: Do we always get segments back from discovery? [We can set](https://github.com/greenkeytech/scribe/blob/master/projects/discovery/src/discovery_utils/prune_lattice.py) `PRUNE_LATTICE=True` and remove this code.

If we require word confusions we can simplify this.
```suggestion
    if "word_confusions" in full_response.get('segments', [""])[0]:
        del full_response["segments"][0]["word_confusions"]
```

**Review by cbrochtrup — CHANGES_REQUESTED** [review]
These fixes look like a bandaid over a more real discovery bug or a misconfiguration in our tests.

It looks like we have inconsistent use of `'` and `"` that we could fix up.

File: testing/evaluate_tests.py [hunk 2]
```diff
@@ -113,6 +113,24 @@ def test_schema(full_response, test_value, test_name=''):
...
+    #   {'had': 1},
+    #    ...
+    #   {'revenue': 1},
+    # and
+    # interpreted_fields:
+    #       company_name: '@nlprocessor_company_name'
+    #       key_metrics:
+    #         revenue:
+    #
+    # the revenue key metric was never found since the word confusion {'revenue':1} was found first
```
[the remark this answers is not in the document]
Comment (sv-github) [reply testing/evaluate_tests.py:128]: :+1: 

I think the main issue was in our _find() process when looking for discovered keys

full_response = {
    'word_confusions': [{'revenue': 1}]
    'interpreted_fields': ['metrics': {'revenue': '$55'}]
}

`_find(full_response, 'revenue')`
will we get the word confusion dict, or will we get the metric?

we dont have a careful way of selecting the correct key

File: testing/evaluate_tests.py (hunk 1 above)
Comment (sv-github) [reply testing/evaluate_tests.py:132]: evaluate_tests.py reads the json test files that are user created

i'd want to keep logic that works with both segments and pruned-lattices

File: testing/evaluate_tests.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbrochtrup) [reply testing/evaluate_tests.py]: Looks like we could prefer keys with values that are strings?

[the remark this answers is not in the document]
Comment (sv-github) [reply testing/evaluate_tests.py]: i think that helps we have strings always in custom schema return_json

File: testing/evaluate_tests.py (hunk 1 above)
Comment (cbrochtrup) [reply testing/evaluate_tests.py]: `PRUNE_LATTICE` [removes the output](https://docs.greenkeytech.com/focus-api/nlp/#configuration_4) word confusions and shouldn't affect the input. Then we can use this variable to remove this code, right? It's clear no tests depend on the output word confusions.

Comment (sv-github) [reply testing/evaluate_tests.py]: i think that would work

discovery would output `full_response` with no word_confusions

Comment (sv-github) [reply testing/evaluate_tests.py]: ahh.. something might not be working as expected for the external_json that we have

the revenue test doesn't pass with PRUNE_LATTICE=True
```
integration test 2          schema              {"revenue": "$3 billion"}                                                                           
                                                {"revenue": 1}
```

**Review by cbrochtrup — APPROVED** [review]
Thanks for the improvements. This is much cleaner and easier to maintain!

**Pull request merged**